Gems/Multiplayer/Code/Source/NetworkEntity/EntityReplication/PropertyPublisher.cpp

```diff
@@ -75,72 +76,83 @@ namespace Multiplayer
         m_replicatorState = EntityReplicatorState::Rebasing;
     }
 
-    void PropertyPublisher::GenerateRecord()
+    void PropertyPublisher::UpdatePendingRecord(NetBindComponent* netBindComponent)
     {
-        AZ_Assert(m_netBindComponent, "NetBindComponent is nullptr");
-        m_netBindComponent->FillReplicationRecord(m_pendingRecord);
+        // Only update the pending record if we don't have a cached delete message already.
+        if (!m_cachedDeleteMessage.GetIsDelete())
+        {
+            AZ_Assert(netBindComponent, "NetBindComponent is nullptr");
+            netBindComponent->FillReplicationRecord(m_pendingRecord);
+        }
     }
 
-    bool PropertyPublisher::HasUpdateEntityRecord()
+    bool PropertyPublisher::HasEntityChangesToSend()
```

Comment:
So as I understand the architecture currently, we have an entity, and that entity has a number of attached replicators. Say for a server with 5 connected clients that can all see the entity, that would be 5 replicators. For each replicator for each frame we're keeping a bitset of properties that represent which properties were modified on that frame. That means for 5 replicators we'd basically have a history of 5 identical bitsets, one copy for each replicator. The only difference being how many entries we keep for each replicator, which depends on the last acked update for each.

Reply:
More specifically, we're keeping a bitset of properties with _unacknowledged_ modifications. If the 5 connected clients all acknowledge at slightly different times, I believe the bitsets can vary from each other on any given frame.